presto-main/src/main/java/io/prestosql/testing/MaterializedResult.java

```diff
@@ -387,7 +388,12 @@ else if (prestoValue instanceof SqlTimeWithTimeZone) {
                         zone);
             }
             else if (prestoValue instanceof SqlTimestamp) {
-                convertedValue = SqlTimestamp.JSON_FORMATTER.parse(prestoValue.toString(), LocalDateTime::from);
+                SqlTimestamp timestampValue = (SqlTimestamp) prestoValue;
+                convertedValue = Timestamps.getInstant(timestampValue.getEpochMicros(), toIntExact(timestampValue.getPicosOfMicros()))
```

Comment:
`Timestamps.getInstant` rounds (by necessity).
It is an appropriate behavior for some cases, but for test code (eg here), we should
be more strict. We should not attempt to use `Instant` where picos matter.

So, I propose you move that method here (or inline) and instead of `roundToNearest`
you do `LongMath.divide(picosOfMicro, PICOSECONDS_PER_NANOSECOND, UNNECESSARY)`